Makefile

```diff
@@ -1,94 +1,75 @@
 .PHONY: test bins clean cover cover_ci
-PROJECT_ROOT = go.uber.org/cadence
-
-export PATH := $(GOPATH)/bin:$(PATH)
-
-THRIFT_GENDIR=.gen
 
 # default target
 default: test
+export PATH := $(GOPATH)/bin:$(PATH)
 
-# define the list of thrift files the service depends on
-# (if you have some)
-THRIFTRW_SRCS = \
-  idl/github.com/uber/cadence/cadence.thrift \
-  idl/github.com/uber/cadence/shared.thrift \
-
-PROGS = cadence-client
+IMPORT_ROOT := go.uber.org/cadence
+THRIFT_GENDIR := .gen/go
+THRIFTRW_SRC := idl/github.com/uber/cadence/cadence.thrift
+# one or more thriftrw-generated file(s), to create / depend on generated code
+THRIFTRW_OUT := $(THRIFT_GENDIR)/cadence/idl.go
 TEST_ARG ?= -race -v -timeout 5m
 BUILD := ./build
 
-THRIFT_GEN=$(GOPATH)/bin/thrift-gen
-
-define thriftrwrule
-THRIFTRW_GEN_SRC += $(THRIFT_GENDIR)/go/$1/$1.go
-
-$(THRIFT_GENDIR)/go/$1/$1.go:: $2
```

Comment:
this file doesn't even exist after generating, which is part of the problem.  I chose a different file to serve as "we ran" - `$((old)THRIFT_GENDIR)/go/cadence/idl.go`